packages/components/src/composite/item.tsx

```diff
@@ -20,9 +20,27 @@ export const CompositeItem = forwardRef<
 	WordPressComponentProps< CompositeItemProps, 'button', false >
 >( function CompositeItem( props, ref ) {
 	const context = useCompositeContext();
+
+	// @ts-expect-error The store prop in undocumented and only used by the
+	// legacy compat layer.
+	const storeViaProps = props.store as Ariakit.CompositeStore;
+	const store = storeViaProps ?? ( context.store as Ariakit.CompositeStore );
+
+	// If the active item is not connected, Composite may end up in a state
+	// where none of the items are tabbable. In this case, we force all items to
+	// be tabbable, so that as soon as an item received focus, it becomes active
+	// and Composite goes back to working as expected.
+	const tabbable = Ariakit.useStoreState( store, ( state ) => {
+		return (
+			state?.activeId !== null &&
+			! store?.item( state?.activeId )?.element?.isConnected
+		);
+	} );
```

Comment:
Good catch 👍 

I wonder if this is something that Ariakit should resolve internally?

Reply:
This fix is already coming after discussing it with ariakit authors in https://github.com/ariakit/ariakit/issues/4129. I think they are cautious about implementing such a fix directly in the library, and instead asked if we wanted to try the fix out in userland first